include/materials/NEMLStress.h

```diff
@@ -15,18 +15,64 @@
 #pragma once
 
 #include "NEMLStressBase.h"
-
 #include "neml_interface.h"
 
+/**
+ * NEMLStress parses the input and xml files and constructs the NEML constitutive model
+ * used by NEMLStressBase.
+ */
+
 class NEMLStress : public NEMLStressBase
 {
 public:
   static InputParameters validParams();
   NEMLStress(const InputParameters & parameters);
 
-protected:
-  /// File name of the NEML XML database
-  FileName _fname;
-  /// Model name in NEML XML database
-  std::string _mname;
+private:
+  /**
+   * Get values from the input file that will be substituted in for the nemlNames provided in the
+   *input file
+   * @param nemlNames variables names from input file
+   * @return vector containing the values that will be substituted in for the each nemlNames
+   **/
+  std::vector<Real> constructNemlSubstitutionList(const std::vector<std::string> & nemlNames) const;
+
+  /**
+   * Error checking: find strings in string list 1 that are not found in string list 2
+   * @param strList1
+   * @param strList2
+   * @return return strings from strList1 not found in strList2
+   **/
+  std::string compareVectorsOfStrings(const std::vector<std::string> & strList1,
+                                      const std::vector<std::string> & strList2) const;
+
+  /**
+   * parse the file into a string
```

Comment:
It kind of does since NEML is going to take this string and run it through an xml parser.  It could parse a file that was just one long string instead of the normal format where XML tags are broken up onto different lines.

Reply:
OK, I'd just change the description to 'Parse the xml file into a string'